pkg/agent/openflow/groups.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 Antrea Authors
```

Comment:
2022

Reply:
> A simple question, why don't change the group id allocator more generally? For example, create different group id allocators using NewGroupAllocator by passing a string argument, like IPv4, IPv6 and others.

We need a unique allocator, since OVS only has one group table where all groups are installed. If we have multiple group id allocators, it is possible that a duplicated ID is assigned.

Reply:
Yes, I know. I think an available way is spit the whole group ID range into several sub ranges. For example, assume that the whole group ID range is from 0-256:
- for IPv4, use the range 0-15
- for IPv6, use the range 16-31
- for next function, use the range 32-63
- for next function, use the range 64-95

The number of sub ranges is decided by the functions. Currently, we only have IPv4 and IPv6, two sub ranges is enough. If we introduce a new function which needs group id allocator, we need four sub ranges.

Reply:
I don't think it is a better idea for your suggested range definition. First, IPv4 and IPv6 groups are not always existing at the same, the only case is for dual-stack case, then it might be a waste for group IDs. Second, 256 might not be enough even for Service (one Service has a single group) even in IPv4 cluster. Third, I don't know your meaning about next function because I can't understand what function is not using IPv4 or IPv6. 

Besides, I think it is different for group ID from flow cookie. It is just an identifier for a group, and we don't see the requirement for a classifier of groups from Antrea code.

Reply:
I don't state what I meant clearly. To be simple, if we introduce another function which needs group id in Antrea, how to get group id from the id allocator?

Reply:
It just uses `v4GroupIDAllocator ` or `v6GroupIDAllocator ` directly, and passes them as a parameter when creating the instance.